text_generation/causal_lm/cpp/continuous_batching/library/src/sequence_group.hpp

```diff
@@ -91,14 +91,14 @@ class Sequence {
 
     // appends new tokens to a generated part
     void append_token(int64_t token_id, float log_prob) {
-        m_cumulative_log_prob += log_prob;
+        m_log_probs.push_back(log_prob);
         m_generated_ids.push_back(token_id);     
     }
 
-    GenerationOutput get_last_generation_output(const GenerationConfig& sampling_params) {
+    GenerationOutput get_last_generation_output() {
         GenerationOutput output;
-        output.score = get_beam_search_score(sampling_params);
-        OPENVINO_ASSERT(m_generated_ids.size());
+        OPENVINO_ASSERT(m_generated_ids.size() && m_log_probs.size());
+        output.score = m_log_probs.back();
```

Comment:
Shouldn't it be cumulative log score? 
Because now `score` means single logprob when we get single output for streaming or cumulative score for complete result, so it might be a bit ambiguous.

Reply:
I thank, the approach is correct. We set `get_cumulative_log_probs()` per seq when it is finished in the case of n seq per seq group.
```
                for (auto& sequence: finished_sequences) {
                    GenerationOutput output;
                    output.generated_token_ids = sequence->get_generated_ids();
                    output.score = sequence->get_cumulative_log_probs();
                    outputs.emplace(sequence->get_grouped_id(), output);
                }
```
But correct for the case of streaming using